packages/polar/src/lib/deploy/contract.ts

```diff
@@ -9,60 +9,173 @@ import {
+import type {
+  Account,
+  AnyJson,
+  ContractFunction,
+  ContractInfo,
+  PolarRuntimeEnvironment
+} from "../../types";
+import { ExecuteResult, getClient, getSigningClient } from "../client";
+import { Abi, AbiParam } from "./abi";
+
+function buildCall (
+  contract: Contract,
+  msgName: string,
+  argNames: AbiParam[]
+): ContractFunction<any> {
+  return async function (
+    ...args: any[]
+  ): Promise<any> {
+    if (args.length !== argNames.length) {
+      console.error(`Invalid ${msgName} call. Argument count ${args.length}, expected ${argNames.length}`);
+      return;
+    }
+
+    const msgArgs: any = {};
+    argNames.forEach((abiParam, i) => {
+      msgArgs[abiParam.name] = args[i];
+    });
+
+    // Query the current count
+    return contract.queryMsg(msgName, msgArgs);
+  };
+}
+
+function buildSend (
+  contract: Contract,
+  msgName: string,
+  argNames: AbiParam[]
+): ContractFunction<any> {
+  return async function (
+    ...args: any[]
+  ): Promise<any> {
+    if (args.length !== argNames.length + 1) {
+      console.error(`Invalid ${msgName} call. Argument count ${args.length}, expected ${argNames.length + 1}`);
+      return;
+    }
+
+    if (
+      args[args.length - 1].address === undefined ||
+      args[args.length - 1].name === undefined ||
+      args[args.length - 1].mnemonic === undefined
+    ) {
+      console.error(`Invalid ${msgName} call. Last argument should be an account object.`);
+      return;
+    }
+
+    const account: Account = (args[args.length - 1] as Account);
+
+    const msgArgs: any = {};
+    argNames.forEach((abiParam, i) => {
+      msgArgs[abiParam.name] = args[i];
+    });
+
+    // Query the current count
+    return contract.executeMsg(msgName, msgArgs, account);
+  };
+}
 
 export class Contract {
   readonly contractName: string;
   readonly contractPath: string;
-  readonly schemaPath: string;
+  readonly initSchemaPath: string;
+  readonly querySchemaPath: string;
+  readonly executeSchemaPath: string;
+  readonly initAbi: Abi;
+  readonly queryAbi: Abi;
+  readonly executeAbi: Abi;
   readonly env: PolarRuntimeEnvironment;
   readonly client: CosmWasmClient;
 
   private codeId: number;
-  private readonly contractCodeHash: string;
+  private contractCodeHash: string;
   private contractAddress: string;
 
+  public query: {
+    [name: string]: ContractFunction<any>
+  };
+
+  public tx: {
+    [name: string]: ContractFunction<any>
+  };
+
   constructor (contractName: string, env: PolarRuntimeEnvironment) {
-    this.contractName = contractName;
+    this.contractName = contractName.replace('-', '_');
     this.codeId = 0;
     this.contractCodeHash = "mock_hash";
-    this.contractAddress = "mock_address";
-    this.contractPath = path.join(ARTIFACTS_DIR, "contracts", `${contractName}.wasm`);
-    this.schemaPath = path.join(SCHEMA_DIR, `${contractName}.json`);
+    this.contractAddress = "secret15l3z73lvy0357qrnyhcfpcxhdrxc209wpvkayl";
```

Comment:
could you please update this to mock_address ?

Reply:
TODO later